vernac/obligations.ml

```diff
@@ -631,6 +631,16 @@ let unfold_entry cst = Hints.HintsUnfoldEntry [EvalConstRef cst]
 let add_hint local prg cst =
   Hints.add_hints local [Id.to_string prg.prg_name] (unfold_entry cst)
 
+let it_mkLambda_or_LetIn_or_clean t ctx =
```

Comment:
Could you add a header comment saying what this function does? I'm not sure to fully understand why we do `if noccurn 1 t then subst1 mkProp t`.